web3/eth.py

```diff
@@ -357,3 +373,26 @@ def contract(self,
 
     def getCompilers(self):
         return self.web3._requestManager.request_blocking("eth_getCompilers", [])
+
+    def _extract_ecdsa_signer(self, msg_hash, signature):
+        msg_hash_bytes = decode_hex(msg_hash) if msg_hash.startswith(b'0x') else msg_hash
```

Comment:
I would prefer we just require that the message be passed in as a hex encoded value.